src/main/java/org/wso2/carbon/inbound/vfs/VFSConsumer.java

```diff
@@ -0,0 +1,749 @@
+                }
+                Map<String, String> queryParams = UriParser.extractQueryParams(fileURI);
+                isMounted = Boolean.parseBoolean(queryParams.get(VFSConstants.IS_MOUNTED));
+                fileObject.setIsMounted(isMounted);
+                wasError = false;
+            } catch (FileSystemException e) {
+                if (retryCount >= vfsConfig.getMaxRetryCount()) {
+                    log.error("Repeatedly failed to resolve the file URI: " + maskURLPassword(fileURI), e);
+                    return null;
+                } else {
+                    log.warn("Failed to resolve the file URI: " + maskURLPassword(fileURI) + ", in attempt "
+                            + retryCount + ", " + e.getMessage() + " Retrying in " + vfsConfig.getReconnectTimeout()
+                            + " milliseconds.");
+                }
+            }
+            if (wasError) {
+                try {
+                    Thread.sleep(vfsConfig.getReconnectTimeout());
+                } catch (InterruptedException e2) {
+                    Thread.currentThread().interrupt();
+                    log.error("Thread was interrupted while waiting to reconnect.", e2);
+                }
+            }
+        }
+        return fileObject;
+    }
+
+    public void close() {
+        isClosed = true;
+        if (retryScheduler != null && !retryScheduler.isShutdown()) {
+            retryScheduler.shutdown();
+            try {
+                if (!retryScheduler.awaitTermination(5, TimeUnit.SECONDS)) {
+                    retryScheduler.shutdownNow();
+                }
+            } catch (InterruptedException e) {
+                retryScheduler.shutdownNow();
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    public void start() {
+        isClosed = false;
+    }
+
+    public void destroy() {
+//        fsManager.close();
+        this.close();
+    }
+
+    private static class ResolvedFileUri {
+        final String resolvedUri;
+        final boolean supportSubDirectories;
+
+        ResolvedFileUri(String uri, boolean subDirs) {
+            this.resolvedUri = uri;
+            this.supportSubDirectories = subDirs;
+        }
+    }
+
+    /**
+     * Task to retry processing of failed records
+     */
+    private class FailedRecordRetryTask implements Runnable {
+        private final FileObject file;
+
+        public FailedRecordRetryTask(FileObject file) {
+            this.file = file;
+        }
+
+        @Override
+        public void run() {
+            try {
+                if (log.isDebugEnabled()) {
+                    log.debug("Retrying failed record: " + maskURLPassword(file.toString()));
+                }
+
+                // Check if file still exists
+                if (file.exists()) {
+                    // Remove from failed records and retry processing
+                    removeFromFailedRecords(file);
+                    processFile(file);
```

Comment:
We don't need to retry failed processing. We only need to retry the failed move action.